policy-controller/core/src/http_route.rs

```diff
@@ -176,3 +197,51 @@ impl PartialEq for QueryParamMatch {
 }
 
 impl Eq for QueryParamMatch {}
+
+impl Status {
+    pub fn collect_from(status: Option<gateway::HttpRouteStatus>) -> Vec<Self> {
+        status
+            .into_iter()
+            .map(|status| {
+                status
+                    .inner
+                    .parents
+                    .iter()
+                    .filter_map(|parent_status| Self::from_parent_status(parent_status))
+                    .collect::<Vec<_>>()
+            })
+            .flatten()
+            .collect()
+    }
+
+    fn from_parent_status(status: &gateway::RouteParentStatus) -> Option<Self> {
+        if let Some(ref kind) = status.parent_ref.kind {
+            if kind != "Server" {
+                return None;
+            }
+        }
+
+        let conditions = status
+            .conditions
+            .iter()
+            .filter_map(|condition| {
+                let type_ = match condition.type_.as_ref() {
+                    "Accepted" => ConditionType::Accepted,
+                    "ResolvedRefs" => ConditionType::ResolvedRefs,
+                    _ => return None,
+                };
+                let status = match condition.status.as_ref() {
+                    "True" => true,
+                    "False" => false,
+                    _ => return None,
+                };
```

Comment:
should these be matched case-insensitively?

Reply:
Yep this would probably be good to do. I'm realizing that I should use `as_str` anyways for these. However, `to_ascii_lowercase` returns a `String`, so then the match cases would need to also be `String`s. Any idea on how to keep this as a `str` match but work with lowercase?